lib/chef/provider/package/chocolatey.rb

```diff
@@ -130,6 +130,17 @@ def remove_package(names, versions)
         # install from, but like the rubygem provider's sources which are more like repos.
         def check_resource_semantics!; end
 
+        def get_choco_version
+          @get_choco_version ||= powershell_exec!("choco --version").result
+        end
+
+        # Choco V2 uses 'Search' for remote repositories and 'List' for local packages
+        def query_command
+          return "list" if get_choco_version.match?(/^1/)
```

Comment:
Just a heads up, search has been an alias for list since 0.9.8.21 (released in November 2013). I think you are safe to use `choco search`. https://docs.chocolatey.org/en-us/choco/release-notes#november-7-2013

- Original Chocolatey CLI was written in PowerShell, before it was written in C#. https://github.com/chocolatey-archive/chocolatey/blob/67fd72e6af03e8ce554ac2d022d9b4a0281e2e43/src/chocolatey.ps1#L203
- It's been in every version of Chocolatey CLI C# - https://github.com/chocolatey/choco/blob/a3186447addfaa2ff410400bc0eed35c64ac3ae3/src/chocolatey/infrastructure.app/commands/ChocolateyListCommand.cs#L28

Reply:
I think something being in the product for nearly 10 years. We we originally going to split the commands for v1, but that moved to v2 to allow for better deprecation warnings among other things. Fun history at https://github.com/chocolatey/choco/issues/158